src/libsystemd-network/sd-ndisc.c

```diff
@@ -701,7 +744,16 @@ _public_ int sd_radv_start(sd_radv *ra) {
                 goto fail;
 
         ra->fd = r;
-        r = 0;
+
+        r = sd_event_add_io(ra->event, &ra->recv_event_source, ra->fd, EPOLLIN, radv_recv, ra);
+        if (r < 0)
+                goto fail;
+
+        r = sd_event_source_set_priority(ra->recv_event_source, ra->event_priority);
+        if (r < 0)
+                goto fail;
+
+        (void) sd_event_source_set_description(ra->recv_event_source, "radv-receive-message");
```

Comment:
Hm, looking at all those repeated calls to `_add_io`, `_set_priority`, `_set_description` and associated error handling, I think we should add an atomic call to do all that. But that's for another PR, as they say.